Pipfile

```diff
@@ -48,6 +48,9 @@ tiktoken = "==0.5.1"
 torch = "==2.1.2"
 tqdm = "==4.65.0"
 xgboost = "==2.0.3"
+sphinx = "==7.2.6"
```

Comment:
Looks like all of those dependencies are fine licensewise. However, one of them depends on `docutils`, which (incorrectly) fails our license check - is there some way to override this check for `docutils`? @felix-zailskas

Reply:
It is `sphinx` causing the issue.

Reply:
Maybe we can fix that by removing sphinx from our Pipfile and installing it separately in the workflow?

Reply:
Let me try, but then we need to keep track of installed version of sphinx manually .

Reply:
Yes, that's not ideal. But considering that we should focus on other things, it is probably an acceptable solution to just always use the latest version.

Reply:
At least it won't break our software if we don't even install it, so worst thing that could happen is probably that the documentation could break at some time due to a sphinx update.

Reply:
Looks good now 👍

Reply:
> The Docutils Public License is designed to be more permissive than some other copyleft licenses, like the GPL. It's intended to ensure that Docutils and any modifications to it remain open-source, but it doesn't extend the requirement to distribute source code to software that merely uses Docutils. This means that using Docutils (or Sphinx, which uses Docutils) to generate your project's documentation doesn't require your project to be open-source or to be distributed under the same license

After reading this, I have put the sphinx under [dev-packages], which seems to solve our problems.

Reply:
Moreover, I think documentation is just as important as the code itself, especially when we hand over it to industry partner .